include/bringauto/modules/ModuleManagerLibraryHandler.hpp

```diff
@@ -31,33 +34,49 @@ class ModuleManagerLibraryHandler {
 	int isDeviceTypeSupported(unsigned int device_type);
 
 	int
-	sendStatusCondition(const struct buffer current_status, const struct buffer new_status, unsigned int device_type);
+	sendStatusCondition(const Buffer current_status, const Buffer new_status, unsigned int device_type);
 
-	int generateCommand(struct buffer *generated_command, const struct buffer new_status,
-						const struct buffer current_status, const struct buffer current_command,
+	int generateCommand(Buffer &generated_command, const Buffer new_status,
+						const Buffer current_status, const Buffer current_command,
 						unsigned int device_type);
 
-	int aggregateStatus(struct buffer *aggregated_status, const struct buffer current_status,
-						const struct buffer new_status, unsigned int device_type);
+	int aggregateStatus(Buffer &aggregated_status, const Buffer current_status,
```

Comment:
Documentation is missing for these functions. for example: There are buffers but what happens when the can is unsuccessful etc.?

Reply:
documentation added